src/josepy/json_util.py

```diff
@@ -457,6 +457,9 @@ def decode_csr(b64der: str) -> josepy.util.ComparableX509:
         raise errors.DeserializationError(error)
 
 
+GenericTypedJSONObjectWithFields = TypeVar('GenericTypedJSONObjectWithFields', bound='TypedJSONObjectWithFields')
```

Comment:
With it's new restricted usage, does this generic still provide value for us? If so, can you give me an example of what that is?

There very well may be a good reason to keep it, but since it's only used on non-self and non-class parameters and on return values, it's not immediately clear to me what its value is anymore.

Reply:
Yes, it is still useful for the `register()` function. It helps mypy to understand that a subclass of `TypedJSONObjectWithFields` annotated with the decorator `@register` is still this subclass, and not `TypedJSONObjectWithFields` itself.